### DISCUSSION TITLE
Automatically reading files changed elsewhere?

### DISCUSSION THREAD

**Discussion by markoa** [zed-industries/zed#6584] [Feature Requests]
Hi, as I'm currently tinkering with Zed I'm of course periodically switching back to my default editor.

If I make some changes & save them to disk in another editor, Zed doesn't pick this up. Instead it flags current buffer as changes that need to be saved.

Frankly this makes it difficult to go back to Zed once I "slip" back to the old editor.

So I'm wondering if there's a setting that I missed or this is not yet implemented?

**Comment by Gantushig77** [comment]
I'm also facing this issue. Zed editor doesn't know when files changed elsewhere. Especially when using gRPC to generate code, it doesn't know whether the old generated files are changed or not. Also, ORM schema files are not syncing. 

Currently using the version Zed 0.146.5 release version.

**Comment by rashidnhm** [comment]
Yup, I keep having to close and re-open the file in Zed.

**Comment by kingtaj** [comment]
With the existing behavior, it can be difficult to tell if there are "local" changes that indicate the file is modified, or if the file was written to elsewhere.

**Comment by ghost** [comment]
+1. When I run linters, they reformat code, and zed won't pick it up. No, running linters "on zed" ain't good, because it'd split brain between "how linters are ran on CI" and "on zed".

**Comment by edencaldas** [comment]
+1 I have been closing and opening zed too much

**Comment by lzhoucs** [comment]
+1

**Comment by dandv** [comment]
@Gantushig77 had filed an issue for watching externally changed files, https://github.com/zed-industries/zed/issues/15791

**Comment by hugodzibetti** [comment]
+1

**Comment by zakony-ai** [comment]
Having same problem, Zed didnt even pickup that I stashed changes and git reseted to origin/mater and was still showing old code. I though I was going insane for a while.

**Comment by Tr3yWay996** [comment]
zed just wiped a good part of my changes made to my rust source code file, this shi is a real deal breaker for me if it cannot do a simple synch since i have my codebase on my external nvme drive for portability. Did changes on my desktop, saved, i just now plugged it back in, added a comment then i saved before realizing it didn't had the changes i did the day earlier, and my who really wanted to daily drive it, wellp...

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
